### DISCUSSION TITLE
On Stale PRs

### DISCUSSION THREAD

**Discussion by ParagEkbote** [huggingface/diffusers#10046] [General]
Hello everyone, I had a query that I needed some guidance. There are 74 PRs with the label 'stable' present in the repository.  I think that nearly half of them are quite close to the finish line before being merged. 

I have tried to tackle this issue in [9940](https://github.com/huggingface/diffusers/pull/9940) and [9935](https://github.com/huggingface/diffusers/pull/9935).

After my fiasco in [9960](https://github.com/huggingface/diffusers/pull/9960), I want to understand what the protocol is for dealing with stale PRs.

Please let me know about the same.

PS: Happy Thanksgiving!

**Comment by yiyixuxu** [comment]
thanks so much for taking the initiative!
I think what you did with #9940 and #9935 was awesome! so feel free to keep doing this:)
we will start to add "contribution welcome" label to these type of stale PRs (close to merge) as well

**Comment by ParagEkbote** [comment]
@yiyixuxu 

For the PRs labeled 'close to merge', do I have to cherry pick commits before submitting my own PR?  Are there any specific guidelines for cherry-picking commits?

**Reply by yiyixuxu** [reply]
ohh these PRs we labeled "close-to-merge" are generally active actually (we just started to use that label), so we should not take them over :)
I'm going through our stale PRs now and will mark these inactive ones as "contribution welcome" so that you can help:) I will also leave comments on specific what type of help is needed (some of the PRs the author are still active and need some help to debug together)

For guidelines to take over someone else's PR, yes, we need to:
1. First, make sure it is okay to take over: we will ask and leave comments on the PR if it is okay! 
2. make sure the original author also gets credit. If you can cherry pick will be great, or you can just manually add them as co-author of the commit 
3. link to the original PR!

**Comment by ParagEkbote** [comment]
Thank you for the guidelines @yiyixuxu 

Looking forward for contributing in the future.
